src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -0,0 +1,217 @@
+    get_bucket_versions: { regular: "s3:ListBucketVersions" },
+    get_bucket_website: { regular: "s3:GetBucketWebsite" },
+    get_bucket_object_lock: { regular: "s3:GetBucketObjectLockConfiguration" },
+    get_bucket: { regular: "s3:ListBucket" },
+    get_object_acl: { regular: "s3:GetObjectAcl" },
+    get_object_tagging: { regular: "s3:GetObjectTagging", versioned: "s3:GetObjectVersionTagging" },
+    get_object_uploadId: { regular: "s3:ListMultipartUploadParts" },
+    get_object_retention: { regular: "s3:GetObjectRetention"},
+    get_object_legal_hold: { regular: "s3:GetObjectLegalHold" },
+    get_object: { regular: "s3:GetObject", versioned: "s3:GetObjectVersion" },
+    get_service: { regular: "s3:ListAllMyBuckets" },
+
+    head_bucket: { regular: "s3:ListBucket" },
+    head_object: { regular: "s3:GetObject", versioned: "s3:GetObjectVersion" },
+
+    post_bucket_delete: { regular: "s3:DeleteObject" },
+    post_object: { regular: "s3:PutObject" },
+    post_object_uploadId: { regular: "s3:PutObject" },
+    post_object_uploads: { regular: "s3:PutObject" },
+    post_object_select: { regular: "s3:GetObject" },
+
+    put_bucket_accelerate: { regular: "s3:PutAccelerateConfiguration" },
+    put_bucket_acl: { regular: "s3:PutBucketAcl" },
+    put_bucket_analytics: { regular: "s3:PutAnalyticsConfiguration" },
+    put_bucket_cors: { regular: "s3:PutBucketCORS" },
+    put_bucket_encryption: { regular: "s3:PutEncryptionConfiguration" },
+    put_bucket_inventory: { regular: "s3:PutInventoryConfiguration" },
+    put_bucket_lifecycle: { regular: "s3:PutLifecycleConfiguration" },
+    put_bucket_logging: { regular: "s3:PutBucketLogging" },
+    put_bucket_metrics: { regular: "s3:PutMetricsConfiguration" },
+    put_bucket_notification: { regular: "s3:PutBucketNotification" },
+    put_bucket_policy: { regular: "s3:PutBucketPolicy" },
+    put_bucket_replication: { regular: "s3:PutReplicationConfiguration" },
+    put_bucket_requestpayment: { regular: "s3:PutBucketRequestPayment" },
+    put_bucket_tagging: { regular: "s3:PutBucketTagging" },
+    put_bucket_versioning: { regular: "s3:PutBucketVersioning" },
+    put_bucket_website: { regular: "s3:PutBucketWebsite" },
+    put_bucket_object_lock: { regular: "s3:PutBucketObjectLockConfiguration" },
+    put_bucket: { regular: "s3:CreateBucket" },
+    put_object_acl: { regular: "s3:PutObjectAcl" },
+    put_object_tagging: { regular: "s3:PutObjectTagging", versioned: "s3:PutObjectVersionTagging" },
+    put_object_uploadId: { regular: "s3:PutObject" },
+    put_object_retention: { regular: "s3:PutObjectRetention" },
+    put_object_legal_hold: { regular: "s3:GetObjectLegalHold"},
+    put_object: { regular: "s3:PutObject" },
+});
+
+const qm_regex = /\?/g;
+const ar_regex = /\*/g;
+
+const predicate_map = {
+    'StringEquals': (request_value, policy_value) => request_value === policy_value,
+    'StringNotEquals': (request_val, policy_value) => request_val !== policy_value,
```

Comment:
please use request_value instead of request_val in all the places in the map.
```suggestion
    'StringNotEquals': (request_value, policy_value) => request_value !== policy_value,
```